jax/_src/xla_bridge.py

```diff
@@ -246,6 +246,52 @@ def make_cpu_client() -> xla_client.Client:
     "cpu", make_cpu_client, priority=0, fail_quietly=False
 )
 
+def _check_ptxas_compat() -> None:
+  import re
+  import jaxlib
+  import subprocess
+
+  from glob import glob
+
+  _jaxlib_path = pathlib.Path(jaxlib.__file__).parent
+  searchable_paths = [_jaxlib_path.parent / "nvidia" / "cuda_nvcc"]
+  searchable_paths += [pathlib.Path(path) for path in os.environ["PATH"].split(":")[:18]]
+  found_ptxas = None
+  for path in searchable_paths:
+      ptxas_relpath = glob("**/ptxas", root_dir=path, recursive=True)
```

Comment:
I don't think you should search recursively: that might be very expensive. You should look only in the search paths exactly.

For the pip package, you might want to add a `/bin`.